src/ProtocolStreamReader.php

```diff
@@ -29,6 +29,10 @@ public function __construct($input)
     {
         $this->input = $input;
         Loop\addReadStream($this->input, function() {
+            if (feof($this->input)) {
+                die;
```

Comment:
Never ever simply die in a function if an error condition happens. Throw an exception, and if it is not caught, it will result in the process exiting with a non-zero exit code.